custom/inddex/couch_db_data_collector.py

```diff
@@ -0,0 +1,133 @@
+
+    @memoized
+    def get_data_from_table(self, table_name=None, fields_and_values: Optional[Tuple] = None, as_dict=False):
+        """
+        filters and collects data from the desired table based on fields_and_values
+
+        :param table_name: name of the table the data is supposed to be collected from
+        :param fields_and_values: tuple of tuples, used for filtering, each tuple should have 2 elements:
+            1st element: column to look for in the data
+            2nd element: tuple of values to look for in the column
+
+            e.g.
+            (
+                ('age', ('7', '14', '21')),
+                ('name', ('Mary', 'Steve'))
+            )
+
+        :param as_dict: determines whether collected data should be returned as dictionary
+        :return: list of filtered data collected from the table OR all rows from the table
+        """
+        data = []
+        if table_name is None and len(self.tables) == 1:
+            table_name = list(self.tables.keys())[0]
+        elif not table_name:
+            raise AttributeError('\'table_name\' must be specified')
+        elif table_name not in self.tables:
+            raise ValueError('Unknown \'table_name\'')
+        records = FixtureDataItem.by_data_type(self.domain, self.tables[table_name])
+
+        if not fields_and_values:
+            return records
+        values = {x[0]: x[1] for x in fields_and_values}
+
+        for record in records:
+            if self._is_record_valid(record, values):
+                if as_dict:
+                    data.append(self.record_to_dict(record))
+                else:
+                    data.append(record)
+
+        return data
+
+    @staticmethod
+    def record_to_dict(record):
+        """
+        transforms :param record from FixtureDataItem object into a dictionary
+        """
+        record_as_dict = {}
+        fields = record.fields
+        for field in fields:
+            field_list = fields[field].field_list
+            value = None if not field_list else field_list[0].field_value
+            record_as_dict[field] = value
+
+        return record_as_dict
+
+    @memoized
+    def get_data_from_table_as_dict(self, key_field, table_name=None, fields_and_values: Optional[Tuple] = None):
+        """
+        does the same filtering as get_data_from_table, but returns the data in
+        form of a dictionary with values of :param key_field as keys
+        """
+        if key_field not in (element[0] for element in fields_and_values):
+            raise ValueError('\'key_field\'s name must be in \'fields_and_values\'')
+        messy_dicts = self.get_data_from_table(
+            table_name=table_name, fields_and_values=fields_and_values, as_dict=True
+        )
+        clean_dicts = {}
+
+        for messy_dict in messy_dicts:
+            clean_dicts[messy_dict[key_field]] = self._get_dict(key_field, messy_dict)
+
+        return clean_dicts
```

Comment:
This is the "dictionary work" I was thinking of:

```suggestion
        data = {}
        for record in self.get_data_from_table(table_name, fields_and_values):
            rdict = self.record_to_dict(record)
            rdict_without_id = {k: v for k, v in rdict.items() if k != key_field} 
            data[rdict[key_field]] = rdict_without_id
        return data
```